source/develop/release-management/features/network/import-vm-net-changes.html.md

```diff
@@ -0,0 +1,164 @@
+                    <name>red</name>
+                </network>
+                <name>gold</name>
+            <source_network_profile>
+            <target_network_profile id=”123”/>
+        </network_mapping>
+        <network_mapping>
+            <source_network_profile>
+                <network>
+                    <name>blue</name>
+                </network>
+                <name>silver</name>
+            <source_network_profile>
+        </network_mapping>
+    <network_mappings>
+    <!-- The new addition end -->
+</action>
+```
+
+Since the mappings are optional, any vNic profile that the mapping isn’t
+ supplied for it, would be mapped to a profile with the same name, if 
+that exists, or an “empty” profile otherwise, like that’s done 
+currently.
+A missing “target_network_profile” element would mean the target profile
+ is an “empty” profile. That is useful for the case that source profile 
+is known to be used in a target system for somewhat else and an 
+alternative profile isn’t exist yet.
+
+#### GUI
+The engine would scan all VMs that are intended to be imported and will 
+provide a list of network+profile that are defined on the vNics of those
+ VMs. Then mappings could be defined (by the user) in an expandable span
+ that would be added to the import VM dialog. 
+Each line would represent a single mapping with the following fields:
+
+* Source network name
+* Target network name
+* Source vNic profile name
+* Target vNic profile name
+
+![Network mapping closed](import_vm-network_mapping_close.png)
+![Network mapping opened](import_vm-network_mapping_open.png)
+
+Initially all vNic profiles would be mapped to a matching (by name) 
+network+profile on the target or would be an “Empty” profile if no match
+ exists. Then the user would be able to change that to any desired 
+profile. The list of the available networks would be filtered according 
+to the chosen cluster and the list of the profiles will be filtered by 
+the chosen network.
+
+
+### Re-assign MAC addresses for the vNics of an imported VM
+The user will be able to request to re-assigning a new MAC instead of 
+one that is problematic in the context of the target cluster.
+The reasons for a MAC to be problematic are:
+
+* Collision - the MAC is owned by an existing vNic.
+* The MAC is out of range of the destination cluster.
+
+Note that a re-assign request modifies only problematic MACs. If a VM 
+has “good” MACs as well, they would remain.
+
+#### REST API
+The import VM request will have additional boolean 
+parameter “reassign_bad_macs”. In order to keep backward compatibility 
+the default value of the new parameter will be False.
+
+```
+POST /storagedomains/{storagedomain:id}/vms/{vm:id}/import
+```
+
+```xml
+<action>
+    <cluster id=”XXX”/>
+    <storage_domain id=”YYY”/>
+    ....
+    <!-- The new addition start -->
+    <reassign_bad_macs>true<reassign_bad_macs>
+    <!-- The new addition end -->
+</action>
+```
+
+#### GUI
```

Comment:
Yes, i can see the 3 images in the link. This 3 are good enough.

On Wed, Sep 14, 2016 at 3:01 PM, dankenigsberg <notifications@github.com>
wrote:

> In source/develop/release-management/features/network/
> import-vm-net-changes.html.md
> <https://github.com/oVirt/ovirt-site/pull/442#discussion_r78734568>:
>
> > +```
> > +POST /storagedomains/{storagedomain:id}/vms/{vm:id}/import
> > +```
> > +
> > +```xml
> > +<action>
> > +    <cluster id=”XXX”/>
> > +    <storage_domain id=”YYY”/>
> > +    ....
> > +    <!-- The new addition start -->
> > +    <reassign_bad_macs>true<reassign_bad_macs>
> > +    <!-- The new addition end -->
> > +</action>
> > +```
> > +
> > +#### GUI
>
> I believe that they exist, just do not show up so nicely in the source
> code page. Can you see 3 images in https://github.com/oVirt/
> ovirt-site/pull/442/files ? what screen are you missing exactly?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/oVirt/ovirt-site/pull/442/files/0468952eda40eaf414601b027ebe69c33a5fe01b#r78734568>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARb3JrvivPNtqO1Mi6xa7nbjR25f-rtcks5qp-IMgaJpZM4Jw78O>
> .
>



-- 
Michael Burman
RedHat Israel, RHEV-M QE Network Team

Mobile: 054-5355725
IRC: mburman